controllers/user_upload.go

```diff
@@ -19,13 +19,14 @@ import (
 	"io"
 	"mime/multipart"
 	"os"
+	"path/filepath"
 
 	"github.com/casdoor/casdoor/object"
 	"github.com/casdoor/casdoor/util"
 )
 
 func saveFile(path string, file *multipart.File) (err error) {
-	f, err := os.Create(path)
+	f, err := os.Create(filepath.Clean(path))
```

Comment:
Does it work for at least Go 1.17?

Reply:
Yes. It works on [go1.17.13](https://pkg.go.dev/path/filepath@go1.17.13#Clean)